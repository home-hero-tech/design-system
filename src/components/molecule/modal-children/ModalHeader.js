import React from 'react';
import PropTypes from 'prop-types';

import css from '../../organism/modal/Modal.module.scss';
import Heading from '../../atom/heading/Heading';
import Text from '../../atom/text/Text';

const ModalHeader = ({ title, description }) => {
  return title || description ? (
    <header className={css['c-modal__heading']}>
      {title ? <Heading level={3} title={title} /> : null}
      {description ? <Text>{description}</Text> : null}
    </header>
  ) : null;
};

ModalHeader.propTypes = {
  description: PropTypes.string,
  title: PropTypes.string
};

ModalHeader.defaultProps = {
  description: null,
  title: null
};

ModalHeader.displayName = 'ModalHeader';

export default ModalHeader;
