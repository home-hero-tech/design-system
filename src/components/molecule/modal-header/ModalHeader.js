import React from 'react';
import PropTypes from 'prop-types';

import css from '../../organism/modal/Modal.module.scss';
import Heading from '../../atom/heading/Heading';
import Text from '../../atom/text/Text';

const ModalHeader = ({ title, description }) => {
  return (
    <header className={css['c-modal__heading']}>
      {title ? <Heading level={3} title={title} /> : null}
      {description ? <Text>{description}</Text> : null}
    </header>
  );
};

ModalHeader.propTypes = {
  description: PropTypes.string,
  title: PropTypes.string
};

ModalHeader.defaultProps = {
  description: null,
  title: 'Header'
};

ModalHeader.displayName = 'ModalHeader';

export default ModalHeader;
