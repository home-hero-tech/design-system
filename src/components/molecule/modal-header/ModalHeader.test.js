import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalHeader from './ModalHeader';

describe('ModalHeader', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <ModalHeader title="Title" description="Description" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
