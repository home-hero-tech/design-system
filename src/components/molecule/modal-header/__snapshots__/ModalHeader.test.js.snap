// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader should match with the snapshot 1`] = `
<header
  className="c-modal__heading"
>
  <Heading
    level={3}
    title="Title"
  />
  <Text
    center={false}
    danger={false}
    end={false}
    fill={false}
    lg={false}
    primary={false}
    sm={false}
    success={false}
    uppercase={false}
    warning={false}
    xs={false}
  >
    Description
  </Text>
</header>
`;
